sunpy/instr/rhessi.py

```diff
@@ -187,42 +184,43 @@ def get_obssum_filename(time_range):
     Returns
     -------
     out : list
-        Returns the filenames of the observation summary file
+        Returns the urls of the observation summary file
 
     Examples
     --------
     >>> import sunpy.instr.rhessi as rhessi
-    >>> rhessi.get_obssum_filename(('2011/04/04', '2011/04/05'))   # doctest: +SKIP
-    ['http://soleil.i4ds.ch/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits']
-
-    .. note::
-        This API is currently limited to providing data from whole days only.
-
+    >>> rhessi.get_observing_summary_filename(('2011/04/04', '2011/04/05'))   # doctest: +SKIP
+    ['https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits',
+     'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110405_031.fits']
     """
     time_range = TimeRange(time_range)
+    # remove time from dates
+    time_range = TimeRange(time_range.start.strftime('%Y/%m/%d'),
+                           time_range.end.strftime('%Y/%m/%d'))
 
-    delta = relativedelta(time_range.end, time_range.start)
-    if delta.years > 0 or delta.months > 0:
-        raise ValueError("Rhessi search results can not be found for a"
-                         " time range crossing multiple months.")
+    filenames = []
 
+    diff_months = (time_range.end.year - time_range.start.year) * 12 +\
+                   time_range.end.month - time_range.start.month
+    first_month = parse_time(time_range.start.strftime('%Y/%m/01'))
+    month_list = list(rrule(MONTHLY, dtstart=first_month, count=diff_months+1))
```

Comment:
Why do you convert it into a list? if that's a generator, then you don't need it for the for loop.